src/components/group-common.js

```diff
@@ -104,76 +106,103 @@ export const NewUserModal = ajaxCaller(class NewUserModal extends Component {
+    const { confirmAddUser, userEmail, roles, suggestions, submitting, submitError, checkingUser } = this.state
+    const emailConstraints = { format: /\w+@(gmail\.com|broadinstitute\.org)$/ }
+    const errors = validate({ userEmail }, { userEmail: emailConstraints })
     const isAdmin = _.includes(adminLabel, roles)
 
-    const userEmailInvalid = !!validate({ userEmail }, { userEmail: { email: true } })
+    const userEmailInvalid = !!validate({ userEmail }, { userEmail: emailConstraints })
     const canAdd = value => value !== userEmail || !userEmailInvalid
 
-    return h(Modal, {
-      onDismiss,
-      title,
-      okButton: h(ButtonPrimary, {
-        tooltip: Utils.summarizeErrors(errors),
-        onClick: () => this.submit(),
-        disabled: errors
-      }, ['Add User'])
-    }, [
-      h(RequiredFormLabel, ['User email']),
-      h(AutocompleteSearch, {
-        autoFocus: true,
-        value: userEmail,
-        onChange: v => this.setState({ userEmail: v }),
-        renderSuggestion: suggestion => div({ style: styles.suggestionContainer }, [
-          div({ style: { flex: 1 } }, [
-            !canAdd(suggestion) && h(TooltipTrigger, {
-              content: 'Not a valid email address'
-            }, [
-              icon('warning-standard', { style: { color: colors.danger(), marginRight: '0.5rem' } })
-            ]),
-            suggestion
-          ])
-        ]),
-        onSuggestionSelected: selection => {
-          this.setState({ userEmail: selection })
-        },
-        onKeyDown: e => {
-          // 27 = Escape
-          if (e.which === 27 && !!userEmail) {
-            this.setState({ userEmail: '' })
-            e.stopPropagation()
+    return Utils.cond(
+      [confirmAddUser, h(Modal, {
+        title: 'This User Is Not Registered',
+        okButton: h(ButtonPrimary, {
+          onClick: async () => {
+            this.setState({ submitting: true })
+            await inviteFunction(userEmail)
+            await this.submit()
           }
-        },
-        suggestions,
-        style: { fontSize: 16 },
-        theme: { suggestion: { padding: 0 } }
-      }),
-      h(FormLabel, ['Role']),
-      h(LabeledCheckbox, {
-        checked: isAdmin,
-        onChange: () => this.setState({ roles: [isAdmin ? userLabel : adminLabel] })
+        }, ['Yes']),
+        cancelText: 'No',
+        onDismiss: () => this.setState({ confirmAddUser: false })
       }, [
-        label({ style: { margin: '0 2rem 0 0.25rem' } }, [`Can manage users (${adminLabel})`])
-      ]),
-      footer && div({ style: { marginTop: '1rem' } }, [footer]),
-      submitError && div({ style: { marginTop: '0.5rem', textAlign: 'right', color: colors.danger() } }, [submitError]),
-      submitting && spinnerOverlay
-    ])
+        'Add ', b(userEmail), ' to the group anyway?',
+        submitting && spinnerOverlay
+      ])],
+      h(Modal, {
```

Comment:
nit: `cond` can accept functions that lazily compute the resulting values, which would avoid doing some unnecessary work building elements for the unused branches.